contrib/mpi-proxy-split/split_process.cpp

```diff
@@ -352,7 +417,9 @@ setLhMemRange()
   close(mapsfd);
   static MemRange_t lh_mem_range;
   if (found) {
-#if !defined(USE_MANA_LH_FIXED_ADDRESS)
+#if defined(CENTOS)
+    dynamicMemRangeHelper(&lh_mem_range, &TWO_GB);
+#elif !defined(USE_MANA_LH_FIXED_ADDRESS)
```

Comment:
Just found MANA already has `readMapsLine`, I don't think you need another function to do the same thing. Why don't you utilize that one?

And you also don't need to define `CENTOS` here. Only `USE_MANA_LH_FIXED_ADDRESS` is good enough.
The function to find the right region should be general.

Reply:
good call. i wrote this before i found those functions, so i based it off of some of the code earlier in the same file. i will change it, and it'll make it a lot more readable as well.